modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscriptionManager.java

```diff
@@ -112,6 +113,12 @@
     private boolean storeUnavailable;
 
     /**
+     * List of suspend request that are received and are not mapped to a subscription. A newly bound subscription must
+     * be suspended if the channel id is in this list.
+     */
+    List<UUID> pendingSuspension;
```

Comment:
Do we really need this list. If channel is not found, let us do nothing. Also handling can be moved to subscriber connection.